### PR TITLE
PS-269: Fix `main.backup_locks` MTR test (8.0)

### PR DESCRIPTION
**Pull request by inikep**
1. Remove `CREATE PROCEDURE` and `CREATE FUNCTION` as they are transactional in 8.0.
2. Add missing `--connection`.

### CODE REVIEW

File: mysql-test/t/backup_locks.test [hunk 1]
```diff
@@ -554,48 +542,21 @@ CREATE DATABASE test1;
 
 --let $id=`SELECT CONNECTION_ID()`
 
---send CREATE PROCEDURE p1() SELECT 1
+--send INSERT INTO t_myisam VALUES(0)
```
Comment (laurynas-biveinis) [inline mysql-test/t/backup_locks.test:541]: This is suspicious - it means that LOCK TABLES FOR BACKUP does not block CREATE PROCEDURE, which is DDL, anymore?

Comment (laurynas-biveinis) [reply mysql-test/t/backup_locks.test:541]: @gl-sergei FYI ^

Comment (inikep) [reply mysql-test/t/backup_locks.test:541]: `--reap` returned without an error with `send CREATE PROCEDURE p1() SELECT 1`.

Comment (laurynas-biveinis) [reply mysql-test/t/backup_locks.test:541]: Hence my question :)

Comment (inikep) [reply mysql-test/t/backup_locks.test:541]: `LOCK TABLES FOR BACKUP` doesn't work for `CREATE PROCEDURE` and `CREATE FUNCTION` anymore as `mysql.proc` (MyISAM) was removed in 8.0 and replaced with `information_schema.routines` (InnoDB).

Comment (laurynas-biveinis) [reply mysql-test/t/backup_locks.test:541]: OK, We have discussed that's it's OK to allow such kinds of DDL under LTFB

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**